src/generate_counterfactual/core.jl

```diff
@@ -53,7 +53,9 @@ function generate_counterfactual(
     decision_threshold::AbstractFloat=0.5,
     gradient_tol::AbstractFloat=parameters[:τ],
     min_success_rate::AbstractFloat=parameters[:min_success_rate],
-    converge_when::Symbol=:decision_threshold,
+    convergence::Union{AbstractConvergence,Symbol}=Convergence.DecisionThresholdConvergence(;
+        decision_threshold=(1 / length(data.y_levels))
+    ),
```

Comment:
Doesn't it always overwrites the `convergence` value?

Reply:
I don't think so - it's the same kind of default argument as the ones other keyword arguments here are using. If the user provides a different `convergence` value, that will be used instead. The tests should also fail if this is always used as the `convergence` value. Am I missing something here?

Reply:
This looks fine to me